addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -328,6 +332,7 @@ function factory(dependencies) {
             inverse: 'attachments',
         }),
         type: attr(),
+        uploadController: attr(),
```

Comment:
Missing documentation.

I also feel upload controller should be auto-created in model when attachment is temporary, and should be auto-removed when it's not a temporary (reasons: coherent code in this model, no coupling with component, more declarative so more robust to future changes, etc.).